src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Elasticsearch/RequestWrapper.cs

```diff
@@ -100,17 +104,42 @@ void InvokeTryProcessResponse(Task responseTask)
             }
         }
 
-        private static void TryProcessResponse(IAgent agent, object response, ITransaction transaction, ISegment segment)
+        private static void ReportError(object apiCallDetails)
+        {
+            var exceptionGetter = _exceptionGetter ??= GetExceptionGetterFromApiCallDetails(apiCallDetails);
+            var ex = exceptionGetter(apiCallDetails);
+
+            if ((ex != null) && (ex is Exception exception))
+            {
+                InternalApi.NoticeError(exception);
```

Comment:
By calling the InternalApi directly here, the error can potentially get missed or reported to the wrong transaction when we are in an async code path. For the async use case it's often safer to hold a reference to the transaction, and call NoticeError directly on the transaction.

Reply:
That's a good point, but the `NoticeError` in `ITransaction` only takes an `Exception`. I'll have to think on a clean way to do it.

Reply:
Yeah, we might want to expand that API so that it can accept non-exception errors.

Reply:
Done.